client.go

```diff
@@ -672,6 +692,40 @@ func (c *Client) Start() (addr net.Addr, err error) {
 	return
 }
 
+// checkProtoVersion verifies that the client is returning a compatible protocol
+// version, and sets the PluginSet that will be used for the negotiated
+// version.
+func (c *Client) checkProtoVersion(protoVersion string) error {
+	serverVersion, err := strconv.Atoi(protoVersion)
+	if err != nil {
+		return fmt.Errorf("Error parsing protocol version %q: %s", protoVersion, err)
+	}
+
+	// record these for the error message
+	var clientVersions []int
+
+	// all versions, including the legacy ProtocolVersion have been added to
+	// the versions set
+	for version := range c.config.VersionedPlugins {
+		clientVersions = append(clientVersions, version)
+
+		if serverVersion != version {
+			continue
+		}
+
+		// set the Plugins value to the compatible set, so the version
+		// doesn't need to be passed through to the ClientProtocol
+		// implementation.
+		c.config.Plugins = c.config.VersionedPlugins[version]
+		c.NegotiatedVersion = version
```

Comment:
This is definitely more of a nit than a true problem, but it took me some time to understand the flow here enough to understand at what point this would get set from the caller's standpoint, and I wonder if it would be slightly clearer to have `checkProtoVersion` _return_ the negotiated version (along with the error value) and assign it to `NegotiatedVersion` up in `Start`.

That way it's slightly clearer which exported function is responsible for setting this, and `checkProtoVersion` can be a pure function as its name would seem to suggest.